components/server/src/workspace/workspace-starter.ts

```diff
@@ -444,6 +444,7 @@ export class WorkspaceStarter {
         region?: string,
     ): Promise<StartWorkspaceResult> {
         const span = TraceContext.startSpan("actuallyStartWorkspace", ctx);
+        span.setTag("region_preference", region);
```

Comment:
I thought it would be interesting to compare the installation with the determined region. So we can look at the cases where it doesn't match. Maybe also logging the country object would help?

Reply:
Do you mean something like 
``` 
desired_region: user-supplied
determined_region: guess
```
?

Reply:
Something that allows us to see how good the region selection matches the current behavior.
Ideally in EU cluster 100% would come out with Europe and likewise 100% in the US cluster end up with NorthAmerica.
If we were logging (or use traces if that works better) the location object and the result we could filter for outliers. Or do you have a different plan to determine the quality of the region selection?

Reply:
I was planning to start by enabling this for myself in staging/prod (the LB header is not testable in preview) and start workspaces with the relevant preferences. We have logs from the proxy which include the GLB header which I was going to use to compare against the traced value.

I think we can include a log directly in the feature flag (and guess) evaluation to make this easier. Will push a change.

Reply:
Added logs. It will now log the header, and request preference, even if the experiment is not enabled. If enabled, it also tells us the outcome of the guess. We can pull this from logs and compare